### ISSUE TITLE
onInit function only called once in life of App

### ISSUE THREAD

**Issue by RodBr** [jonataslaw/get#203]
**Describe the bug**
Moving to a page that uses `onInit `to do some initial functionality works first time only. 

**To Reproduce**
1. On navigating to a new page, controller is created in GetBuilder.
2. Controller does data retrieval in `onInit` method.
3. Leave page, and controller is deleted from memory
4. Return to page, and controller is recreated, but `onInit `is not run so data is not retrieved.

**Expected behavior**
onInit is run each time a controller is instantiated.

**Flutter Version:**
1.17.3

**Get Version:**
2.11.0

**Minimal reproduce code**
```
 @override
  void onInit() {
    getDataFromFirebase();
    super.onInit();
  }
```

**Comment by jonataslaw** [comment]
> **Describe the bug**
> Moving to a page that uses `onInit `to do some initial functionality works first time only.
> 
> **To Reproduce**
> 
> 1. On navigating to a new page, controller is created in GetBuilder.
> 2. Controller does data retrieval in `onInit` method.
> 3. Leave page, and controller is deleted from memory
> 4. Return to page, and controller is recreated, but `onInit `is not run so data is not retrieved.
> 
> **Expected behavior**
> onInit is run each time a controller is instantiated.
> 
> **Flutter Version:**
> 1.17.3
> 
> **Get Version:**
> 2.11.0
> 
> **Minimal reproduce code**
> 
> ```
>  @override
>   void onInit() {
>     getDataFromFirebase();
>     super.onInit();
>   }
> ```

I tested it on the pub dev sample application and is working with SmartManagement.full.
Which option are you using?

**Comment by jonataslaw** [comment]
I think you are using SmartManagement.onlyBuilder.
I'm going to take a look at that now.

**Comment by RodBr** [comment]
I am using SmartManagement.onlyBuilder.

**Comment by RodBr** [comment]
isDependencyInit is not firing because the controller is still kept in the tree even though I used Get.offAndToNamed and the controller was removed from memory.

**Comment by jemariads** [comment]
I have same problem on Get 2.11.0, Controller does data retrieval in onInit method. and the smartManagement is default, I didn't set it.

**Comment by jonataslaw** [comment]
Fix on 2.11.1
